### ISSUE TITLE
Event listeners propagating where they shouldn't

### ISSUE THREAD

**Issue by fardog** [jarofghosts/dom-delegation-stream#2]
An example of the issue I'm seeing:

``` javascript
// file test.js
var events = require('dom-delegation-stream')

var el1 = document.createElement('div')
var button1 = document.createElement('input')
button1.type = 'button'
button1.name = 'button1'

el1.appendChild(button1)

var el2 = document.createElement('div')
var button2 = document.createElement('input')
button2.type = 'button'
button2.name = 'button2'

el2.appendChild(button2)

document.body.appendChild(el1)
document.body.appendChild(el2)

events(el1, 'click', 'input[type=button]').on('data', function(e) {
  console.log('element1', e.target.name)
})

events(el2, 'click', 'input[type=button]').on('data', function(e) {
  console.log('element2', e.target.name)
})

button1.click()
button2.click()
```

run with `browserify test.js | jsdom-eval`

output:

```
element2 button1
element1 button1
element2 button2
element1 button2
```

expected:

```
element1 button1
element2 button2
```

**Comment by fardog** [comment]
Note that the same results are acheived with:

``` javascript
events(el1, 'click').on('data', function(e) {
  console.log('element1', e.target.name)
})

events(el2, 'click').on('data', function(e) {
  console.log('element2', e.target.name)
})
```

**Comment by fardog** [comment]
resolved in #3 

thanks!
